### PR TITLE
feature (refs T29879): add permission contracts and logic

### PR DESCRIPTION
**Pull request by dresslerdemos**

### CODE REVIEW

**Review by graupnerdemos — APPROVED** [review]
As a non-blocking aside: I think we should get into the habit of always adding class / interface documentation blocks in demosplan-addon. Even if only to add the `@since` annotation for API documentation purposes.

File: src/Permission/CorePermissionEvaluatorInterface.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+declare(strict_types=1);
+
+namespace DemosEurope\DemosplanAddon\Permission;
+
+/**
+ * Marker interface to allow for dependency injection.
+ *
+ * To be implemented by the core only!
+ */
+interface CorePermissionEvaluatorInterface extends PermissionEvaluatorInterface
```
Comment (graupnerdemos) [inline src/Permission/CorePermissionEvaluatorInterface.php:12]: Looking at this again this does not have to be here. Doesn't hurt either however imo.

File: src/Permission/PermissionEvaluatorInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     * Will return `false` if the permission is unknown or not accessible by the caller.
+     *
+     * @param non-empty-string $permissionName
+     */
+    public function isPermissionEnabled(string $permissionName): bool;
+
+    /**
+     * @param non-empty-string $permissionName
+     */
+    public function isPermissionKnown(string $permissionName): bool;
```
Comment (graupnerdemos) [inline src/Permission/PermissionEvaluatorInterface.php:24]: Can you please add a short describing sentence of the methods' purpose? I don't quite understand yet if that method has uses outside the Permissions-class set or is only part of the internal evaluation process.

File: src/Permission/CorePermissionEvaluatorInterface.php (hunk 1 above)
Comment (dresslerdemos) [reply src/Permission/CorePermissionEvaluatorInterface.php:12]: If it isn't necessary I'd like to remove it, but I don't see how else I'd be able to inject `Permissions.php` into `AbstractPermissionEvaluator`.

File: src/Permission/PermissionEvaluatorInterface.php (hunk 2 above)
Comment (dresslerdemos) [reply src/Permission/PermissionEvaluatorInterface.php:29]: Does this help? e1076a60ae5febcdf8b0adb2f5bc417c7a4dc8a9

**Comment by dresslerdemos** [comment]
changes after initial approve were verbally approved

**Pull request merged**